extensions/amp-video-iframe/0.1/amp-video-iframe.js

```diff
@@ -266,12 +268,36 @@ class AmpVideoIframe extends AMP.BaseElement {
       return;
     }
 
+    if (eventReceived == 'analytics') {
+      this.dispatchCustomAnalyticsEvent_(data['eventType'], data['vars']);
+      return;
+    }
+
     if (ALLOWED_EVENTS.indexOf(eventReceived) > -1) {
       this.element.dispatchCustomEvent(eventReceived);
       return;
     }
   }
 
+  /**
+   * @param {string} eventType
+   * @param {!Object<string, string>=} vars
+   */
+  dispatchCustomAnalyticsEvent_(eventType, vars = {}) {
+    const requiredEventTypePrefix = 'video-custom-';
+
+    user().assertString(eventType, '`eventType` missing in analytics event');
+
+    user().assert(
+        (new RegExp(`^${requiredEventTypePrefix}`)).test(eventType),
```

Comment:
nit: It's inefficient to use the regex constructor for a constant string because the regex engine has to rebuild the regex each time this function is called. Consider creating the regex once as a literal `/^video-custom-/` and caching the regex outside the class .